bitCoinHash.js

```diff
@@ -16,25 +17,11 @@ function getBlockURL(blockId) {
 	return 'https://blockexplorer.com/api/block/' + blockId;
 }
 
-function getBlockFromWeb(url, https) {
-	var Promise = require('promise');
-
-	return new Promise(function(fullfill, reject) {
-		var concat = require('concat-stream');
-
-		https.get(url, function(res) {
-			var concatStream = concat(function(buffer) {
-				fullfill(JSON.parse(buffer.toString()));
-			});
-
-			res.on('error', function(err) {
-				reject(err);
-			});
-			res.pipe(concatStream);
-		}).on('error', function(e) {
-			reject(e);
-		});	
-	});
+function getBlockFromWeb(url) {
+	return rp(url)
+		.then(function (body) {
+			return JSON.parse(body);
```

Comment:
Any reason to have the function wrapper?

```js
.then(function (body) { JSON.parse(body); });
```

is equivalent to -

```js
.then(JSON.parse);
```

Assuming that the method is context (i.e. `this`) independent.